Domain.Tests/EventSourcedAggregateTests.cs

```diff
@@ -404,6 +443,55 @@ public TestCommandPreconditionVerifier(Func<bool> hasBeenApplied)
             {
                 return hasBeenApplied();
             }
+        }
+
+        [Category("Performance")]
+        [Test]
+        public void When_calling_ctor_of_an_aggregate_with_a_large_number_of_source_events_in_non_incrementing_order_then_the_operation_completes_quickly()
+        {
+            var count = 1000000;
+            var largeListOfEvents = Enumerable.Range(1, count).Select(i => new PerfTestAggregate.SimpleEvent { SequenceNumber = i }).ToList();
+            Shuffle(largeListOfEvents, new Random(42));
+
+            var sw = Stopwatch.StartNew();
+            var t = new PerfTestAggregate(Guid.NewGuid(), largeListOfEvents);
+            sw.Stop();
+
+            Console.WriteLine("Elapsed: {0}ms", sw.ElapsedMilliseconds);
+            t.Version.Should().Be(count);
+            t.NumberOfUpdatesExecuted.Should().Be(count);
+            sw.Elapsed.Should().BeLessThan(TimeSpan.FromSeconds(10));
+        }

```

Comment:
About how many seconds does it take this test to run? The 10 second elapsed time suggests this can be a very slow test.